### ISSUE TITLE
False positive E1102 when adding a non-callable attribute to object

### ISSUE THREAD

**Issue by exquo** [PyCQA/pylint#4221]
### Steps to reproduce

```python
'test'.startswith('t')
o = object()
x = 42
if o is x:
    o.startswith = None
```

### Current behavior

```
$ pylint -E file.py
************* Module file
file.py:1:0: E1102: 'test'.startswith is not callable (not-callable)
```

### Expected behavior
No errors.

### pylint --version output

```
pylint 2.7.2
astroid 2.5.1
Python 3.7.3 (default, Jul 25 2020, 13:03:44)
[GCC 8.3.0]
```

Pylint probably reasons like this: `str` is a subclass of `object`; `object`'s attribute is assigned to non-callable; thus, `str`'s attribute by the same name should be non-callable. I can see the logic here, but the error is reported on the very first line, before the attribute has the chance to get re-assigned. 

By itself, the above example code is not very meaningful. I'm reporting it because this seems to be the underlying cause of https://github.com/PyCQA/pylint/issues/3970#issuecomment-794286592. Also, it might manifest itself in other, more practical scenarios than the example above.
